samples/02.branding-styling-and-customization/g.send-timeout/README.md

```diff
@@ -0,0 +1,309 @@
+    );
++ }
+
++ window.addEventListener('hashchange', renderWebChat);
+
++ renderWebChat();
+
+  document.querySelector('#webchat > *').focus();
+```
+
+Then, we will read the `sendTimeout` value from `location.hash` and update Web Chat.
+
+```diff
+  const res = await fetch('https://webchat-mockbot.azurewebsites.net/directline/token', { method: 'POST' });
+  const { token } = await res.json();
+  const directLine = window.WebChat.createDirectLine({ token });
+
++ function getHash() {
++   const hash = new URLSearchParams((location.hash || '').substr(1));
++   const hashJSON = {};
++
++   for (const [key, value] of hash.entries()) {
++     hashJSON[key] = value;
++   }
++
++   return hashJSON;
++ }
+
+  function renderWebChat() {
++   const { retry } = getHash();
++   const sendTimeout =
++     retry === 'fn'
++       ? activity => {
++           console.group('Running custom function to evaluate send timeout for activity.');
++           console.log(activity);
++           console.log('Returning 2 seconds.');
++           console.groupEnd();
++
++           return 2000;
++         }
++       : retry === 'false'
++       ? false
++       : +retry || undefined;
++
++   console.log(
++     `Using send timeout value of ${
++       typeof sendTimeout === 'function' ? 'custom function' : JSON.stringify(sendTimeout)
++     }`
++   );
+
+    window.WebChat.renderWebChat(
+      {
+-       directLine
++       directLine,
++       styleOptions: {
++         sendTimeout
++       }
+      },
+      document.getElementById('webchat')
+    );
+  }
+
+  window.addEventListener('hashchange', renderWebChat);
+
+  renderWebChat();
+
+  document.querySelector('#webchat > *').focus();
+```
+
+## Completed code
+
+Here is the finished `index.html`:
+
+```html
+<!DOCTYPE html>
+<html lang="en-US">
+  <head>
+    <title>Web Chat: Customize send timeout</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <script src="https://cdn.botframework.com/botframework-webchat/latest/webchat.js"></script>
+    <style>
+      html,
+      body {
+        height: 100%;
+      }
+      body {
+        margin: 0;
+      }
+
+      #webchat {
+        height: 100%;
+        width: 100%;
+      }
+
+      #buttonBar {
+        display: flex;
+        flex-wrap: wrap;
+        left: 10px;
+        margin: 0;
+        position: absolute;
+        top: 10px;
+      }
+
+      #buttonBar > a {
+        background-color: White;
+        border: solid 2px #0063b1;
+        color: #0063b1;
+        font-family: Calibri, 'Helvetica Neue', Arial, sans-serif;
+        font-size: 80%;
+        margin: 0 10px 10px 0;
+        padding: 5px 8px;
+        text-decoration: none;
+      }
+
+      #buttonBar > a.selected {
+        background-color: #0063b1;
+        color: White;
+      }
+    </style>
+  </head>
+  <body>
+    <div id="webchat" role="main"></div>
+    <p id="buttonBar">
+      <a href="#retry=default">Default retry</a>
```

Comment:
I don't like the idea of using hyperlinks as buttons. Can we change to buttons?

Reply:
This is legit hyperlinks. When clicked, it is going to change the URL and it will trigger our code without page refresh.